claripy/backends/backend_z3.py

```diff
@@ -45,6 +45,16 @@ def _add_memory_pressure(p):
 
 supports_fp = hasattr(z3, 'fpEQ')
 
+#
+# Per-thread Z3 solver
+#
+
+reuse_z3_solver = True if os.environ.get('REUSE_Z3_SOLVER', "False").lower() in {"1", "true", "yes", "y"} else False
+
+
+class Z3ThreadSolvers:
```

Comment:
Why the dedicated class here, instead of just sticking it into BackendZ3's tls instance?

Reply:
Done.